### PR TITLE
Mrce branch

### PR DESCRIPTION
**Pull request by QB3**
branch to add MRCE algorithm

### CODE REVIEW

**Comment by agramfort** [comment]
what are you trying to do?

**Comment by QB3** [comment]
I am trying to make the CI test pass.
Pytest always runs in python 2 mode, which breaks the tests.
I am trying to make the tests pass by running pytest with python 3.

File: clar/duality_gap.py [hunk 1]
```diff
@@ -2,11 +2,29 @@
 
 from numba import njit
 from numpy.linalg import norm
+from numpy.linalg import slogdet
 
 from clar.utils import l_2_inf
 from clar.utils import l_2_1
 
 
+def get_p_obj_MRCE(X, Y, Y2, Sigma, Sigma_inv, alpha, alpha_Sigma_inv, B, sigma_min):
```
Comment (agramfort) [inline clar/duality_gap.py]: don't put capital letters in function names

File: clar/tests/test_mrce.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import numpy as np
+from clar.solvers import solver, update_Sigma_glasso
+from clar.utils import (
+    get_alpha_max, get_sigma_min)
+from clar.data.artificial import get_data_me
+# import matplotlib.pyplot as plt
+
+
+def test_update_Sigma_glasso():
```
Comment (agramfort) [inline clar/tests/test_mrce.py]: Sigma -> sigma

File: clar/tests/test_mrce.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    # plt.figure()
+    # plt.imshow(S_star @ S_star.T)
+    # plt.show(block=True)
+    # plt.figure()
+    # plt.imshow(precision)
+    # plt.show(block=True)
+    # import ipdb; ipdb.set_trace()
+
+
+def test_MRCE():
```
Comment (agramfort) [inline clar/tests/test_mrce.py]: test_MRCE -> test_mrce

File: README.rst [hunk 4]
```diff
@@ -6,7 +6,7 @@ CLaR
 This package implements the CLaR alorithm, a fast algorithm to handle sparse linear regression with heteroscedastic noise, see:
 Bertrand, Q., Massias, M., Gramfort, A., & Salmon, J. (2019). Concomitant Lasso with Repetitions (CLaR): beyond averaging multiple realizations of heteroscedastic noise. arXiv preprint arXiv:1902.02509.
 
-It also implements a variation of the MRCE algorithm with a L21 penalization on the regression coefficient Beta, and which take in account the repetitions see
+It also implements a variation of the mrce algorithm with a L21 penalization on the regression coefficient Beta, and which take in account the repetitions see
```
Comment (agramfort) [inline README.rst]: not here :)

File: clar/data/artificial.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    rng = check_random_state(seed)
+
+    X = get_dictionary(
+        dictionary_type, n_channels=n_channels,
+        n_sources=n_sources, rho=rho, meg=meg, eeg=eeg, seed=seed )
+    S_star = get_S_star(
+        noise_type=noise_type, n_channels=n_channels,
+        rho_noise=rho_noise, meg=meg, eeg=eeg, seed=seed)
+
+    rng=check_random_state(seed)
```
Comment (agramfort) [inline clar/data/artificial.py]: pep8

File: .travis.yml [hunk 6]
```diff
@@ -1,23 +1,27 @@
...
   - pip install flake8 check-manifest
   - pip install codecov
   - python setup.py build_ext --inplace
   - pip install -e .
 script:
+  - source activate CLaR-env
+  # - python3.6 -m pytest
+  - pytest
   # - pytest -lv --cov-report term-missing celer --cov=celer --cov-config .coveragerc
   - flake8 --count celer
```
Comment (agramfort) [inline .travis.yml]: this should be:

flake8 --count clar

  - flake8 --count celer
```
```suggestion
  - flake8 --count celer
```
```suggestion
  - flake8 --count celer
```

File: .travis.yml [hunk 7]
```diff
@@ -1,23 +1,27 @@
...
+  - pip install pytest-env
   - pip install flake8 check-manifest
   - pip install codecov
   - python setup.py build_ext --inplace
   - pip install -e .
 script:
+  - source activate CLaR-env
+  # - python3.6 -m pytest
+  - pytest
   # - pytest -lv --cov-report term-missing celer --cov=celer --cov-config .coveragerc
```
Comment (agramfort) [inline .travis.yml]: remove this line

File: .travis.yml [hunk 8]
```diff
@@ -1,23 +1,27 @@
...
+  - python --version
   - conda install --yes --quiet pytest pytest-cov coverage
+  - pip install pytest-env
   - pip install flake8 check-manifest
   - pip install codecov
   - python setup.py build_ext --inplace
   - pip install -e .
 script:
+  - source activate CLaR-env
+  # - python3.6 -m pytest
```
Comment (agramfort) [inline .travis.yml]: remove

File: .travis.yml [hunk 9]
```diff
@@ -1,23 +1,27 @@
...
+  - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda3.sh
+  - chmod +x miniconda3.sh
+  - ./miniconda3.sh -b -p /home/travis/miniconda3
+  - export PATH=/home/travis/miniconda3/bin:$PATH
   - conda update --yes --quiet conda
 install:
-  - conda env create --file environment.yml python=${PYTHON_VERSION}
+  - conda env create --file environment.yml python=3.6
   - source activate CLaR-env
+  - python --version
```
Comment (agramfort) [inline .travis.yml]: remove

**Comment by mathurinm** [comment]
@QB3 fixed travis in antoher PR. You need to rebase

Try to keep separate goals for separate PR: you are working on travis and MRCE here.

File: clar/tests/test_mrce.py [hunk 10]
```diff
@@ -1,9 +1,9 @@
 import numpy as np
-from sgcl.solvers import solver, update_Sigma_glasso
-from sgcl.utils import (
+from clar.solvers import solver, update_Sigma_glasso
+from clar.utils import (
     get_alpha_max, get_sigma_min)
 from data.artificial import get_data_me
-import matplotlib.pyplot as plt
+# import matplotlib.pyplot as plt
```
Comment (mathurinm) [inline clar/tests/test_mrce.py]: remove all commented lines

**Pull request merged**